### ISSUE TITLE
Flatpak build

### ISSUE THREAD

**Issue by Blzut3** [ECWolfEngine/ECWolf#317]
**[Original report](https://bitbucket.org/ecwolf/ecwolf/issues/317) by John Doe (Bitbucket: [John Doe](https://bitbucket.org/John Doe), ).**

----------------------------------------

Plase make a [flathub.org](http://flathub.org) build, this will help in installing and auto-updating ECWolf on linux distros other than Ubuntu.

**Comment by Blzut3** [comment]
Not personally interested, but someone else can contribute one if they wish.

**Comment by Blzut3** [comment]
**Original comment by Jose Valecillos (Bitbucket: [Jose Valecillos](https://bitbucket.org/Jose Valecillos), ).**

----------------------------------------

Hello Branden,

Sorry for commenting on a closed issue, but I figured it made sense to follow up here instead of opening a new one.

I put together a Flatpak manifest for ECWolf, tested it, and it seems to work well. Would you be okay with me submitting it to Flathub? I’m happy to:

* ‌    Share it with you so you can take ownership,
* ‌    Co-maintain it with you, or
* ‌    Maintain it myself if you’d rather not be involved—totally up to you!

Let me know what you think. Thanks!

Regards,

Jose

‌

**Comment by Blzut3** [comment]
You’re welcome to create and maintain the package.  If you want to add me to the repo for bus factor reasons that’s fine, but my personal interest in Flatpaks hasn’t changed so I will expect you to handle maintenance.

By the way, since it’s probably not well known that I own the domain, for the macOS app the identifier is org.ecwolf.ECWolf.  You should probably use that for the Flatpak as well.

**Comment by Blzut3** [comment]
**Original comment by Jose Valecillos (Bitbucket: [Jose Valecillos](https://bitbucket.org/Jose Valecillos), ).**

----------------------------------------

Hello again,

Let me share a quick update. I[ submitted the package to FlatHub](https://github.com/flathub/flathub/pull/6230) and the reviewers asked for a required[ Metadata Info](https://docs.flathub.org/docs/for-app-authors/requirements#required-metadata) file \(a[ Freedesktop standard](https://www.freedesktop.org/software/appstream/docs/) for application info\) to be pulled from upstream \(this repository\). So I want to check with you if you have any problem for me including that in the docs folder along with the man pages. You can see the content of the file here:[ org.ecwolf.ECWolf.metainfo.xml](https://github.com/flathub/flathub/blob/3c559ce37709777e37b5d48cbb1c9d5b0c83ff16/org.ecwolf.ECWolf.metainfo.xml).

On the other hand, in my manifest I was moving the binary file from “`games`” into “`bin`” following the _Arch Linux_ package for _ECWolf_. The reviewer asked about it, so I tried to pass `CMAKE_INSTALL_BINDIR` to **CMake** to change it \(instead of doing a post install step\). And that’s where I noticed a potential bug where the variable is set in such a way that the value cannot be changed. I think it might be a bug, and ~~after struggling with how cumbersome is Bitbucket,~~ I opened a PR for that: [https://bitbucket.org/ecwolf/ecwolf/pull-requests/22](https://bitbucket.org/ecwolf/ecwolf/pull-requests/22)

Please let me know your thoughts.

Best Regards,

**Comment by Blzut3** [comment]
Yes, it is fine to submit the metadata info or any other files that should live in this repo.

By the way you have the license as BSD 3 clause which is inaccurate.  The resulting binary, when using the default compile options, is licensed as GPLv2\+.  ECWolf requires that new code in the project is liberally licensed due to code sharing with the then non-GPL ZDoom.  Still today we share code with Raze which has a complicated licensing situation.  Additionally with the GPL option turned off you can build a binary licensed under the original non-free EULA, although there aren’t a ton of reasons you’d want to do this.  Either way this detail is mostly irrelevant to packaging.

**Comment by Blzut3** [comment]
**Original comment by Jose Valecillos (Bitbucket: [Jose Valecillos](https://bitbucket.org/Jose Valecillos), ).**

----------------------------------------

Thank you for merging my two previous PRs on this topic. I have one additional question regarding the next release.

To maintain cleaner submissions, it would be beneficial to have a new tag/release. This would eliminate the need to add Metadata Info or the `CMAKE_INSTALL_BINDIR` fix manually or through a patch. I've already added a patch for gcc14 compilation, although this is already resolved on master. Do you have any thoughts about a possible next release? Since it could be just a bump to the next patch version I supposed but I’m not so sure what has changed since 1.4.1.

**Comment by Blzut3** [comment]
Not a ton.  The maint branch has what’s staged for the next release \(haven’t picked these changes back yet\).  Not sure at this second when the next release will happen, but I really need to get around to it.  Was going to figure that out today but some personal stuff came up today.

**Comment by Blzut3** [comment]
**Original comment by Jose Valecillos (Bitbucket: [Jose Valecillos](https://bitbucket.org/Jose Valecillos), ).**

----------------------------------------

No problem. Whenever you do, please update if you can the entry for the revision history in the in the metainfo file: [docs/org.ecwolf.ECWolf.metainfo.xml#lines-50](https://bitbucket.org/ecwolf/ecwolf/src/2c64179560f74ddb3b489938c8eea5276f32bb72/docs/org.ecwolf.ECWolf.metainfo.xml#lines-50)

That is something I only thought after the fact.

**Comment by Blzut3** [comment]
**Original comment by Jose Valecillos (Bitbucket: [Jose Valecillos](https://bitbucket.org/Jose Valecillos), ).**

----------------------------------------

Hello @{557058:d7312301-2c30-4af2-8c0b-9e8966e2fced} . Have you had the change to check my suggestions about having a new release? It could be just a minor release just to reflect the metainfo changes and the potential submission to Flathub. Think about it and let me know.

**Comment by Blzut3** [comment]
Thanks for the reminder.  Will try to get to this soon.

**Comment by Blzut3** [comment]
Got the big items I had in mind done.  Likely will finish up next weekend.  Need to check the macOS build for regressions and see if there’s some low hanging fruit bugs to clean up.

**Comment by Blzut3** [comment]
1\.4.2 is tagged.  Working through all the builds.

**Comment by Blzut3** [comment]
@jvalecillos Not sure if you're still interested in maintaining the Flatpak, but wanted to notify you that ECWolf has moved to GitHub. (Announcement: https://maniacsvault.net/entry94)

**Comment by PerAstraAdDeum** [comment]
Hi there,

may I ask what the current state of the Flatpak is? Couldn't find in on Flathub?

Best regards

**Comment by Blzut3** [comment]
The PR to follow would be https://github.com/flathub/flathub/pull/6230

Besides the acknowledgement of my last post here, I've not heard anything.

**Comment by PerAstraAdDeum** [comment]
> The PR to follow would be [flathub/flathub#6230](https://github.com/flathub/flathub/pull/6230)
> 
> Besides the acknowledgement of my last post here, I've not heard anything.

Thanks for the info! I'll leave a comment and follow the issue there.
